<launch>

  <arg name="config" default="$(find kal_trajectory_planner_ros_tool)/launch/params/trajectory_planner_parameters.yaml" />
  <arg name="required" default="false"/>
  <arg name="node_name" default="trajectory_planner"/>
  <arg name="map_frame"/>
  <arg name="vehicle_frame"/>
  <arg name="path_topic"/>
  <arg name="trajectory_topic"/>


  <!-- this is just to be compatible with the nodelet version -->
  <arg name="nodelet_manager_name" default="nodelet_manager_trajectory_planner"/>
  <arg name="start_nodelet_manager" default="true"/>


  <node pkg="kal_trajectory_planner_ros_tool" type="trajectory_planner" name="$(arg node_name)" ns="/planning" output="screen">
    <rosparam command="load" file="$(arg config)"/>
    <param name="/map_frame" value="$(arg map_frame)" />
    <param name="/vehicle_frame" value="$(arg vehicle_frame)" />
    <param name="/path_subscriber_topic" value="$(arg path_topic)" />
    <param name="/trajectory_publisher_topic" value="$(arg trajectory_topic)" />
  </node>

</launch>
